src/backend/mi2/mi2.ts

```diff
@@ -138,7 +138,9 @@ export class MI2 extends EventEmitter implements IBackend {
                 if (parsed.token !== undefined) {
                     if (this.handlers[parsed.token]) {
                         this.handlers[parsed.token](parsed);
-                        delete this.handlers[parsed.token];
+                        if (parsed.resultRecords) {
+                            delete this.handlers[parsed.token];
+                        }
                         handled = true;
                     }
                 }
```

Comment:
Yes, but perhaps it's best to re-arrange this logic a bit to resolve both your comments. Since we're just ignoring responses without a resultRecord in the handler, we might as well not pass it to the handler at all, but instead extend the `if (parsed.token !== undefined)` check on line 138 to `if (parsed.token !== undefined && parsed.resultRecords)`. 
This way, the handler can be simplified, and the handling in this part isn't as ambiguous.

Reply:
Made this change, it looks a lot cleaner, IMO, and removes the ambiguity you're pointing out in these two comments